pandas/_libs/tslib.pyx

```diff
@@ -3436,7 +3436,24 @@ def cast_to_nanoseconds(ndarray arr):
     return result
 
 
-def pydt_to_i8(object pydt):
+cdef inline _to_i8(object val):
```

Comment:
you need a return type (int64_t)

Reply:
There are cases in which it returns the `object` input.